Improve culling starting point without selection

Solarer opened the pull request:
Bugfix: When entering culling mode without a selection and without mouseover, culling will be reset to the beginning of the collection ignoring the previous offset position.
This is due to always passing `-1` instead of `lib->thumbtable_offset` as offset parameter in file `src/views/lighttable.c`.

Also, if `dt_culling_init` is called with a starting position, mouseover was being ignored.

TurboGit commented:
Please never merge master in a topic branch, you need to rebase on master instead. Also, when you feel the PR is ready you need to squash the commits together and avoid some non useful intermediate state. Why? Because when using a bisect all those are noise and even worst could introduce bugs or being not compilable.

Solarer commented:
@TurboGit sorry, is there anything I can do now after pushing? I probably should have squashed/rebased before.
What about branches that I work on for a longer time and that other people might have pulled in the meantime? I was always told not to rebase if other people collaborate on the stuff.

TurboGit commented:
Yes, here is what you can do:
```
$ git checkout improve_culling_starting_point
$ git rebase master
```

This will remove the merge commit. For the squashing you can use an interactive rebase:

`$ git rebase --interactive master`

Then, a window will open with information about the way to squash, move, remove, reword each individual commit.

If something goes wrong (conflicts):

`$ git rebase --abort`

If you are unsure before the rebase keep a copy of the current branch status:

`$ git branch keep-culling improve_culling_starting_point`

Then after the interactive rebase you can check that the branch is fully equivalent.

`$ git diff improve_culling_starting_point keep-culling`

Solarer commented:
Rebased and squashed.
I think there is something still missing - don't merge it to master yet. I need get the list of images that are blinking in filemanager when leaving culling without selection. Any idea what I have to look for?

AlicVB commented:
@Solarer : the css class for that is added to the thumbnails here : https://github.com/darktable-org/darktable/blob/702fec9e186e387457a2987a7e277a77b06d8c42/src/dtgtk/thumbtable.c#L2035
they are related to `darktable.view_manager->active_images`

AlicVB reviewed (changes requested):
Thanks for this fix. Can I let you do the same for full preview ?
When you are at it, in dt_culling_init, I would rename offset to something like fallback_offset so it's more clear ? (I can do it if you prefer)

Solarer commented on src/views/lighttable.c:
```diff
@@ -212,7 +212,7 @@ static void _lighttable_check_layout(dt_view_t *self)
     }
     else
-      dt_culling_init(lib->culling, -1);
+      dt_culling_init(lib->culling, lib->thumbtable_offset);
```
done

Solarer commented:
> they are related to `darktable.view_manager->active_images`

Thanks, unfortunately I have no experience with CSS and as far as I can tell `darktable.view_manager->active_images` is empty when I call it from within `dt_culling_init`. Can I somehow retrieve the dt_last_active image ids from the CSS or some other place?

AlicVB reviewed (approved):
All good for me now.
Thanks for the cleanup too !
